lib/actions/deploy.coffee

```diff
@@ -61,23 +65,36 @@ performUpload = (image, token, username, url, size, appName, logStreams) ->
 
 	uploadToPromise(post, size, logStreams)
 
+uploadLogs = (logs, token, url, buildId, username, appName) ->
+	request = require('request')
+	request.post
+		url: getBuilderLogPushEndpoint(url, buildId, username, appName)
+		auth:
+			bearer: token
+		body: Buffer.from(logs).toString('base64')
+
 uploadToPromise = (request, size, logStreams) ->
 	logging = require('../utils/logging')
+
 	new Promise (resolve, reject) ->
 
 		handleMessage = (data) ->
 			data = data.toString()
 			logging.logDebug(logStreams, "Received data: #{data}")
 
-			obj = JSON.parse(data)
-			if obj.type?
-				switch obj.type
-					when 'error' then reject(new Error("Remote error: #{obj.error}"))
-					when 'success' then resolve(obj.image)
-					when 'status' then logging.logInfo(logStreams, "Remote: #{obj.message}")
-					else reject(new Error("Received unexpected reply from remote: #{data}"))
-			else
-				reject(new Error("Received unexpected reply from remote: #{data}"))
+			Promise.try ->
```

Comment:
I think this should be a normal try/catch, having this weird hybrid `Promise.try` inside a `new Promise` is really weird